File: mobile/app/src/main/java/com/smartsuschef/mobile/data/models/ForecastData.kt
package com.smartsuschef.mobile.data.models

data class ForecastData(
    val id: String,
    val storeId: Int,
    val recipeId: String,
    val forecastDate: String,
    val predictedQuantity: Int,
    val createdAt: String,
    val updatedAt: String,
)

File: backend/SmartSusChef.Api/Models/ForecastData.cs
namespace SmartSusChef.Api.Models;

public class ForecastData
{
    public Guid Id { get; set; }

    // Linked to a specific Store for multi-tenant isolation
    public int StoreId { get; set; }

    // The specific Recipe being predicted
    public Guid RecipeId { get; set; }

    public DateTime ForecastDate { get; set; }

    public int PredictedQuantity { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public Store Store { get; set; } = null!;
    public Recipe Recipe { get; set; } = null!;
}